cuda_core/cuda/core/experimental/_kernel_arg_handler.pyx

```diff
@@ -85,6 +113,8 @@ cdef inline int prepare_arg(
         (<supported_type*>ptr)[0] = cpp_complex.complex[float](arg.real, arg.imag)
     elif supported_type is cpp_double_complex:
         (<supported_type*>ptr)[0] = cpp_complex.complex[double](arg.real, arg.imag)
+    elif supported_type is __half_raw:
+        (<supported_type*>ptr).x = <int16_t>(arg.view(numpy_int16))
     else:
         (<supported_type*>ptr)[0] = <supported_type>(arg)
```

Comment:
Here is the bug: When `arg` is a `np.float16` scalar, the old code would treat it as `int16_t` (due to lack of standard C++ identifier for fp16 before C++23), and the scalar would be `static_cast` to `int16_t`, which triggered non-trivial conversion operators. The new code ensures that there is no conversion and the bytes are `reinterpret_cast`'d.